asset/modules/notesLib/authenticator.py

```diff
@@ -0,0 +1,72 @@
+import logging
+
+from pydrive.auth import GoogleAuth
+from pydrive.drive import GoogleDrive
+
+logging.getLogger('googleapiclient.discovery_cache').setLevel(logging.ERROR)
+
+def driveAuthorization():
+    """Authorizes access to the Google Drive"""
+    
+    gauth = GoogleAuth()
+    gauth.LocalWebserverAuth() # Creates local webserver and auto handles authentication.
+    drive = GoogleDrive(gauth)
+    return drive
+
+def getGrootFolderId(drive):
+    """Shows the Folder Id of Groot Folder in your Google Drive"""
+
+    fileFolderList = drive.ListFile({'q': "'root' in parents and trashed=false"}).GetList()
+    for GrootFolder in fileFolderList:
+        if GrootFolder['title'] == 'Groot':       
+            print ('title: %s, id: %s' % (GrootFolder['title'], GrootFolder['id']))
+            return GrootFolder['id']
+
+def getFileList(drive):
+    """Gives the file list inside of the Groot Folder"""
+    
+    file_list = drive.ListFile({'q': "'11--DsLz-mGaNvL0nOzhUeQ7RmMcV-BZC' in parents and trashed=false"}).GetList()
```

Comment:
@iamchiragsharma, what is this key or query in this line?
```python
{'q': "'11--DsLz-mGaNvL0nOzhUeQ7RmMcV-BZC' in parents and trashed=false"}
```
I mean how is this working? My first guess is that it is some kind of key?

Reply:
Sorry I made a mistake. I have replaced it with the right code. I mistakenly pasted my folder id instead of retrieving it. It's fixed now.

Reply:
This line searches for the folder Groot which is a parent folder for all the files.

Reply:
I have pushed to the respective pull request for notes.

[Refer here](https://github.com/ugroot/GROOT/pull/38)